src/sentry/workflow_engine/migrations/0032_migrate_metric_alerts.py

```diff
@@ -0,0 +1,468 @@
+        if not snuba_query:
+            logger.info("alert rule missing snuba query", extra={"alert_rule_id": alert_rule.id})
+            continue
+        try:
+            query_subscription = QuerySubscription.objects.get(snuba_query=snuba_query.id)
+        except QuerySubscription.DoesNotExist:
+            logger.info(
+                "query subscription does not exist", extra={"snuba_query_id": snuba_query.id}
+            )
+            continue
+        data_source = DataSource.objects.create_or_update(
+            organization_id=organization_id,
+            source_id=str(query_subscription.id),
+            type="snuba_query_subscription",
+        )
+
+        # create DCG
+        data_condition_group = DataConditionGroup.objects.create_or_update(
+            organization_id=organization_id,
+        )
+        # create detector
+        detector = Detector.objects.create_or_update(
+            project_id=project.id,
+            enabled=enabled,
+            created_by_id=create_activity.user_id,
+            name=alert_rule.name,
+            workflow_condition_group=data_condition_group,
+            type=MetricAlertFire.slug,
+            description=alert_rule.description,
+            owner_user_id=alert_rule.user_id,
+            owner_team=alert_rule.team,
+            config={
+                "threshold_period": alert_rule.threshold_period,
+                "sensitivity": alert_rule.sensitivity,
+                "seasonality": alert_rule.seasonality,
+                "comparison_delta": alert_rule.comparison_delta,
+                "detection_type": alert_rule.detection_type,
+            },
+        )
+        # create workflow
+        workflow = Workflow.objects.create_or_update(
+            name=alert_rule.name,
+            organization_id=organization_id,
+            when_condition_group=None,
+            enabled=enabled,
+            created_by_id=create_activity.user_id,
+            config={},
+        )
+
+        open_incident = Incident.objects.get_active_incident(alert_rule, project)
+        if open_incident:
+            state = (
+                DetectorPriorityLevel.MEDIUM
+                if open_incident.status == IncidentStatus.WARNING.value
+                else DetectorPriorityLevel.HIGH
+            )
+        else:
+            state = DetectorPriorityLevel.OK
+
+        data_source.detectors.set([detector])
+        # create detector state
+        DetectorState.objects.create_or_update(
+            detector=detector,
+            active=False,
+            state=state,
+        )
+        # create lookup tables
+        AlertRuleDetector.objects.create_or_update(alert_rule=alert_rule, detector=detector)
+        alert_rule_workflow = AlertRuleWorkflow.objects.create_or_update(
+            alert_rule=alert_rule, workflow=workflow
+        )
+        DetectorWorkflow.objects.create_or_update(detector=detector, workflow=workflow)
+
+        # migrate triggers
```

Comment:
It might be nice to pull this out into another function for ease of readability and testing